packages/client/src/adapters/facts/interface.ts

```diff
@@ -83,7 +84,7 @@ export interface TableExportResult {
   message: string;
 }
 export default interface NaviFactAdapter {
-  fetchDataForRequest(request: Request, options: RequestOptions): Promise<unknown>;
+  fetchDataForRequest(request: Request, options: RequestOptions): Promise<unknown> | Task<unknown>;
   urlForFindQuery(request: Request, options: RequestOptions): string;
-  urlForDownloadQuery(request: Request, options: RequestOptions): Promise<string>;
+  urlForDownloadQuery(request: Request, options: RequestOptions): Promise<string> | Task<unknown>;
```

Comment:
What is the Task generic here? Should it be `string` like the promise?

Reply:
Yep, it should be string
```suggestion
  urlForDownloadQuery(request: Request, options: RequestOptions): Promise<string> | Task<string>;
```